### PR TITLE
Contract tests

### LINKED ISSUE #10: Write tests for agreement.sol
Agreement.sol has no tests...
You should write some before deploying this.

### PR DESCRIPTION
**Pull request by mehtaculous**

### CODE REVIEW

**Review by stinkyfi — APPROVED** [review]

**Comment by auryn-macmillan** [comment]
Sorry guys, I'm not really suited to review front-end code, unfortunately.

**Comment by samwexus** [comment]
@auryn-macmillan hey, sorry I didn't mean to include the frontend updates in this PR. 

the main file to look at is the `Agreement.js` test

**Comment by mehtaculous** [comment]
@auryn-macmillan Hey, sorry didn't mean to include the frontend updates in this PR.

the main file to look at is `test/Agreement.js`

File: test/Agreement.js [hunk 1]
```diff
@@ -0,0 +1,82 @@
...
+      expect(admin.address).to.equal(owner);
+      expect(memberRole).to.equal(true);
+    });
+  });
+
+  describe("optInToDAO", function () {
+    it("should require the contract to be active", async function () {
+      await contract.activationSwitch();
+
+      try {
```
Comment (auryn-macmillan) [inline test/Agreement.js:30]: I believe you can do this like this to make it a little cleaner

```js
expect(
  await contract.optInToDAO()
).to.be.revertedWith("The contract no longer allows opt-ins");
```

**Review by auryn-macmillan — APPROVED** [review]

**Pull request merged**